### PR TITLE
TGR-111 upgrade dependencies

### PR DESCRIPTION
**Pull request by croyfish**
## JIRA tickets:
- [Recap Hold Request Consumer - Ruby + GHA update](https://newyorkpubliclibrary.atlassian.net/browse/TGR-111)

## Things Done:
- Updated Ruby / Gems
- Implemented GitHub Actions for tests
- Implemented GitHub Actions for deployments
- Set `qa` as the default branch and added branch protection rules to `qa` and `production`

## How to Test:

- Build the app locally via README instructions.
- Run the unit tests

### CODE REVIEW

**Comment by croyfish** [comment]
I think the Gemfile.lock can be pared down if we figure out which aws services are needed from the sdk

File: provisioning/base/resources.tf [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  bucket = "nypl-github-actions-builds-${var.environment}"
+  key    = "recap-hold-request-consumer-${var.environment}-dist.zip"
+  acl    = "private"
+  source = data.archive_file.lambda_zip.output_path
+  etag   = filemd5(data.archive_file.lambda_zip.output_path)
+}
+
+# Create the lambda:
+resource "aws_lambda_function" "lambda_instance" {
+  description   = "Lambda that sits behind API Gateway to serve as a proxy for arbitrary endpoints that we want to make asynchronous."
```
Comment (danamansana) [inline provisioning/base/resources.tf:35]: Is this description accurate?

**Review by nonword — APPROVED** [review]

File: provisioning/base/resources.tf [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  function_name = "RecapHoldRequestConsumer-${var.environment}"
+  handler       = "index.handler"
```
Comment (danamansana) [inline provisioning/base/resources.tf:37]: is this accurate? It looks like the handler might be `main.handle_event`?

**Review by danamansana — CHANGES_REQUESTED** [review]
Looks like a couple of values may have been copied over mistakenly? Otherwise looks good

File: provisioning/base/resources.tf (hunk 1 above)
Comment (croyfish) [reply provisioning/base/resources.tf:35]: 🤷‍♂️ Probably not

Comment (danamansana) [reply provisioning/base/resources.tf:35]: I've been taking `description` from the top line of the README, in this case something like `This is a consumer for handling hold requests from ReCAP for NYPL and partner patrons.`

**Review by danamansana — APPROVED** [review]

**Review by charmingduchess — APPROVED** [review]
My Ruby environment is a little screwy so I can't run this, but changes LGTM!

**Pull request merged**